src/MoBi.Presentation/DTO/EventAssignmentBuilderDTO.cs

```diff
@@ -17,6 +21,73 @@ public EventAssignmentBuilderDTO(EventAssignmentBuilder eventAssignmentBuilder)
       {
          _eventAssignmentBuilder = eventAssignmentBuilder;
          Rules.Add(createTargetHasToBeSetRule);
+         Rules.Add(cannotAssignCircularRefRuleUseAsValue);
+         Rules.Add(cannotAssignCircularRefRuleAssignment);
+         Rules.Add(cannotAssignCircularRefRuleFormula);
+      }
+
+      private IBusinessRule cannotAssignCircularRefRuleFormula { get; } = CreateRule.For<EventAssignmentBuilderDTO>()
+         .Property(x => x.NewFormula)
+         .WithRule((dto, newFormula) => !formulaRefersToAssignmentFormula(dto, newFormula))
+         .WithError(AppConstants.Validation.CannotAssignAFormulaThatReferencesTheAssignmentTarget);
+
+      private static bool formulaRefersToAssignmentFormula(EventAssignmentBuilderDTO dto, FormulaBuilderDTO formulaDTO)
+      {
+         var assignmentBuilder = dto._eventAssignmentBuilder;
+         var assignmentPath = assignmentBuilder.ObjectPath;
+         var formulaPaths = formulaDTO.ObjectPaths.Select(x => x.FormulaUsablePath).ToList();
+
+         if (!canReferToAssignmentTarget(assignmentBuilder.UseAsValue, assignmentPath, formulaPaths))
+            return false;
+
+         return formulaPathsReferToAssignmentTarget(assignmentPath, assignmentBuilder, formulaPaths);
+      }
```

Comment:
minor issue: the 3 `formulaRefersToXYZ `functions could be simplified a bit by passing directly `dto._eventAssignmentBuilder` as the 1st argument and extracting the last 3 code lines into a separate function.